src/libs/io/io.c

```diff
@@ -9,13 +9,97 @@
 
 #include <kdbhelper.h>
 #include <kdbio.h>
+#include <kdbioplugin.h>
 #include <kdbioprivate.h>
 #include <kdblogger.h>
 #include <kdbprivate.h>
 
+#include <stdio.h>
+
+/**
+ * @internal
+ * Retrieves a function exported by a plugin.
+ *
+ * @param  plugin Plugin handle
+ * @param  name   Function name
+ * @return        Pointer to function
+ */
+static size_t getPluginFunction (Plugin * plugin, const char * name)
```

Comment:
@markus2330: Is there a way to share functions like these internally (e.g. with libraries and plugins) and not export them? I feel like I'm missing some obvious choice here :)

Reply:
To share a library function (in any way outside of the source file/library itself) requires the function to be exported (including the c file is a hack only suitable for tests).

To share a function in a plugin you can use the contract (the `exports`). Of course you could realize something similar for your library but that most likely is overkill.

Btw. do you know elektraInvokeGetFunction from the elektra-invoke library implemented in src/libs/invoke/invoke.c?
It looks very similar to your code.

Reply:
Ok, thank you very much!

Yes I have seen it, but it uses `ElektraInvokeHandle` which wraps `Plugin`. I'm using it at various points (like when [iterating global plugins](https://github.com/ElektraInitiative/libelektra/pull/1825/files#diff-c7a393ddf5f92626daa6ec9e73774309R94) or [passing the I/O binding to list's subplugins](https://github.com/ElektraInitiative/libelektra/pull/1825/files#diff-2233b84e0123b41a7f3eca5fa59dec4dR363) and in the notification library to access register*-functions) where I directly have a Plugin handle available.

How about adding this particular function to `libplugin` as `elektraPluginGetFunction`?

Reply:
Yes, this is a very good idea!

I am even thinking about making elektra-invoke dependent on elektra-plugin to use your function. Both getPluginFunction and elektraInvokeGetFunction are lacking the optimization to directly use plugin->kdb* when requesting one of the 5 standard functions (open, close, get, set, error).

If we add the dep to elektra-plugin from elektra-invoke we could get rid of any access to Plugin's struct within elektra-invoke, making elektra-invoke independent of internals of the Plugin's struct. elektra-plugin obviously needs by design access to the struct.

Reply:
Ok, great! I will add it to libplugin. 

I can make the changes to elektra-invoke later on within this PR.